app/phone/src/main/java/dev/jdtech/jellyfin/PlayerActivity.kt

```diff
@@ -382,4 +420,72 @@ class PlayerActivity : BasePlayerActivity() {
+        sleepJob = CoroutineScope(Dispatchers.Main).launch {
+            delay(timerDuration)
+            binding.playerView.player?.playWhenReady = !binding.playerView.player?.playWhenReady!!
+            binding.playerView.findViewById<FrameLayout>(R.id.player_controls).visibility = View.GONE
+            binding.playerView.findViewById<FrameLayout>(R.id.sleep_mode_view).visibility = View.VISIBLE
+            window.clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
+            isControlsLocked = true
+        }
+        Toast.makeText(applicationContext, formatTimerMessage(timerDuration), Toast.LENGTH_SHORT).show()
+    }
+
+    private fun showSleepTimerDurationDialog() {
+        val timerTexts =
+            listOf("1 minute", "5 minutes", "10 minutes", "30 minutes", "1 hour", "2 hours")
+        val durationValues = longArrayOf(60000L, 300000L, 600000L, 1800000L, 3600000L, 7200000L)
+        var selectedDuration = durationValues[2] // Default: 10 minutes in milliseconds
+
+        val builder = MaterialAlertDialogBuilder(this)
+        builder.setTitle("Select Sleep Timer Duration")
+            .setSingleChoiceItems(
+                timerTexts.toTypedArray(),
+                timerTexts.indexOf("${selectedDuration / 60000} minutes"),
+            ) { _, which ->
+                selectedDuration = durationValues[which]
+            }
+            .setNegativeButton("Cancel") { dialog, _ ->
+                dialog.dismiss()
+                wasDialogShown = false
+            }
+            .setPositiveButton("OK") { dialog, _ ->
+                // Start the sleep timer with the selected duration
+                startSleepTimer(selectedDuration)
+                isSleepModeEnabled = true
+                dialog.dismiss()
+            }
+            .setCancelable(true)
+        wasDialogShown = true
+        builder.create().show()
+    }
+
+    private fun disableSleepMode() {
+        binding.playerView.player?.playWhenReady = true
+        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
+        binding.playerView.findViewById<FrameLayout>(R.id.sleep_mode_view).visibility = View.GONE
+        binding.playerView.findViewById<FrameLayout>(R.id.player_controls).visibility = View.VISIBLE
+        isControlsLocked = false
+        isSleepModeEnabled = false
+    }
+
+    private fun formatTimerMessage(timerDuration: Long): String {
+        val hours = timerDuration / 3600000L
+        val minutes = (timerDuration % 3600000L) / 60000L
+
+        val hourText = if (hours == 1L) "hour" else "hours"
+        val minuteText = if (minutes == 1L) "minute" else "minutes"
+
+        return when {
+            hours > 0 && minutes > 0 ->
+                "Sleep Mode will be enabled in $hours $hourText and $minutes $minuteText"
+            hours > 0 ->
+                "Sleep Mode will be enabled in $hours $hourText"
+            else ->
+                "Sleep Mode will be enabled in $minutes $minuteText"
```

Comment:
Moving this as a string resource will be better for Translations.

Reply:
> Moving this to a string resource will be better for Translations.

I think I moved everything to a string resource, but I had to use @SuppressLint("StringFormatMatches") for that function because I couldn't figure out what the thing wanted from me. It works fine tho.

Reply:
@AstroTheAstronaut `StringFormatMatches` lint is probably because you used `%1$s` while you are passing a number value which expects `%1$d` in some places like `hours`, `minutes`, etc.